Documentation/Development.md

```diff
@@ -6,27 +6,232 @@
 
 * [Swift for TensorFlow toolchain][toolchain].
 * An environment that can run the Swift for TensorFlow toolchains: Ubuntu 18.04, macOS with Xcode 10, or Windows 10.
+* CMake.  CMake 3.16 or newer is required to build with CMake.
+
+### Building
+
+### X10
+
+X10 provides the underlying lazy tensor implementation that is used for the
+tensor computation in the TensorFlow Swift APIs. This library builds atop of
+the TensorFlow library, using the XLA compiler to perform optimizations over
+the tensor computation.
+
+#### (Option 1) Use a prebuilt version of X10
+
+You can use a prebuilt version of the X10 library for building the TensorFlow
+Swift APIs package if you do not need to make changes to the X10 library
+implementation.
+
+There are prebuilt versions of the X10 library for certain platforms. If a
+prebuilt is unavailable for your desired platform, you can build X10 from
+source following the instructions below.
+
+- [Windows 10 (x64)](https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC80Mzc2OC9hcnRpZmFjdE5hbWUvdGVuc29yZmxvdy13aW5kb3dzLXg2NA2/content?format=zip)
+- [macOS (x64)](https://artprodeus21.artifacts.visualstudio.com/A8fd008a0-56bc-482c-ba46-67f9425510be/3133d6ab-80a8-4996-ac4f-03df25cd3224/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2NvbXBuZXJkL3Byb2plY3RJZC8zMTMzZDZhYi04MGE4LTQ5OTYtYWM0Zi0wM2RmMjVjZDMyMjQvYnVpbGRJZC80Mzc2OC9hcnRpZmFjdE5hbWUvdGVuc29yZmxvdy1kYXJ3aW4teDY00/content?format=zip)
+
+The location where you extract the prebuilt library is important to remember
+as it will be required in building the TensorFlow Swift APIs.
+
+#### (Option 2) Building X10
+
+#### Requirements
+
 * Bazel. This can be installed [manually][bazel] or with
 [Bazelisk][bazelisk]. You will need a version supported by TensorFlow
 (between `_TF_MIN_BAZEL_VERSION` and `_TF_MAX_BAZEL_VERSION` as specified in
 [tensorflow/configure.py][configure.py]).
 * Python3 with [numpy][numpy].
-* CMake.  CMake 3.16 or newer is required to build with CMake.
 
-### Building and testing
+#### Building
+
+The X10 implementation is distributed as part of the TensorFlow Swift APIs
+repository. It consists of two halves:
+
+1. [XLA Client](Sources/x10/xla_client)
+1. [X10](Sources/x10/xla_tensor)
```

Comment:
Do you intentionally use `1.` for all ordered list items, and count on the Markdown renderer to show the correct number? I would be more comfortable using the explicit count `2.` - elsewhere in the file too.

Could you please add a short description of the purpose of these two parts?
```suggestion
1. [XLA Client](Sources/x10/xla_client): the XLA client <insert description>
2. [X10](Sources/x10/xla_tensor): <insert description>
```